Tryouts/Plugins/Services/MockAuthentication/MockAuthenticationService.cs

```diff
@@ -0,0 +1,75 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using NP.Utilities;
+using NP.Utilities.Attributes;
+using NP.Utilities.BasicServices;
+
+namespace MockAuthentication;
+
+[Implements(typeof(IAuthenticationService), IsSingleton = true)]
+public class MockAuthenticationService : VMBase, IAuthenticationService
+{
+    [Part]
+    private ILog? Log { get; set; }
+
+    private string? _currentUserName;
+    public string? CurrentUserName
+    {
+        get => _currentUserName;
+        private set
+        {
+            if (_currentUserName == value)
+                return;
+
+            _currentUserName = value;
+
+            OnPropertyChanged(nameof(CurrentUserName));
+            OnPropertyChanged(nameof(IsAuthenticated));
+        }
+    }
+
+    // Is authenticated is true if and only if the CurrentUserName is not zero
+    public bool IsAuthenticated => CurrentUserName != null;
+
+    public bool Authenticate(string userName, string password)
+    {
+        if (IsAuthenticated)
+        {
+            throw new Exception("Already Authenticated");
+        }
+
+        CurrentUserName =
+                (userName == "nick" && password == "1234") ? userName : null;
+
+        if (IsAuthenticated)
+        {
+            Log?.Log
```

Comment:
I do not think we established the coding guidelines yet. In general I prefer to spread the code vertically, not horizontally, so that I'd be able to see the code by moving the vertical scroll bar and not the horizontal one.

Reply:
I think I wasn't clear enough in my question again 😄.  I wasn't talking about horizontal placement vs. vertical placement.  I agree that the need for horizontal scrolling in a code file should be avoided.  We can talk about why that is valuable, but that's a separate conversation from the one I was trying to have.  I was actually referring to the specifics of formatting used when a statement is split over multiple lines.  We can get into the details of specific rules, but let me try to convey it first just using examples.  You wrote this:
```
Log?.Log
(
    LogKind.Info,
    nameof(IAuthenticationService),
    $"Authenticated user '{userName}'");
```
Based on the guidelines I'm used to from projects I've worked on I would have expected to see either this:
```
Log?.Log(
    LogKind.Info,
    nameof(IAuthenticationService),
    $"Authenticated user '{userName}'");
```
or this
```
Log?.Log(LogKind.Info,
         nameof(IAuthenticationService),
         $"Authenticated user '{userName}'");
```
Note that of the two forms of formatting I recorded I prefer the former.  This is because it creates better consistency, always using the common indenting level.  The latter indents to the level that the first parameter is written on the same line of the method name, which can creates much inconsistency.

Reply:
Removed